### ISSUE TITLE
[NFR] Implement a tree move method to allow root nodes to become branches of existing tree

### ISSUE THREAD

**Issue by temuri416** [phalcon/incubator#535]
Hi,

**Given**:

1. A structure with two independent roots (one tree does not know anything about another):
![image](https://cloud.githubusercontent.com/assets/3987203/12699499/3d90ab24-c78b-11e5-9f15-d5793f4ec19e.png)

2. DB rows look like:
![image](https://cloud.githubusercontent.com/assets/3987203/12699505/6ea6d436-c78b-11e5-81a4-d968097630f9.png)

**Required**:

An ability to move **Root2** or any of its children _anywhere_ inside the tree whose root is **Root1**.

This would be an opposite of ```moveAsRoot()``` - importing a (sub)tree from the outside anywhere inside the current tree.

Having this feature would complete the ability to assemble or break up trees.

Thanks!

**Comment by sergeyklay** [comment]
@temuri416 
Could you please take a look at the `NestedSetTest::testShouldMoveNodesBetweenTrees` #552
This is exactly what you had in mind?

**Comment by temuri416** [comment]
@sergeyklay 
Pardon my ignorance, but how can I check out contents of pull request?

**Comment by sergeyklay** [comment]
:neutral_face:

Just add to the your `composer.json`
```json
    "repositories": [
        {
            "type": "vcs",
            "url": "git@github.com:sergeyklay/incubator.git"
        }
    ]
```

and then run `composer update`. Of course you'll need use `require` like this:
```json
{
    "require": {
        "phalcon/incubator": "2.0.x-dev"
    }
}
```

See: https://getcomposer.org/doc/05-repositories.md#repository

**Comment by temuri416** [comment]
The following works as expected on #552:

```php

// Create Root #1.
$root = new Node;
$root->title = 'ROOT A';
$root->saveNode();
$r1 = $root->getId();

// Add "A" as child of root
$node1=new Node;
$node1->title='A';
$node1->appendTo($root);

// Add "B" as last child of root - should be below A
$node2=new Node;
$node2->title='B';
$node2->appendTo($root);

// Add "C" as first child of root - should be above A
$node3=new Node;
$node3->title='C';
$node3->prependTo($root);

// Create Root #2.
$root = new Node;
$root->title = 'ROOT 2';
$root->saveNode();
$r2 = $root->getId();

// Create child of Root #2
$node1=new Node;
$node1->title='2.a';
$node1->prependTo($root);

// Load Root #1.
$root1 = Node::findFirst($r1);

// Load Root #2.
$root2 = Node::findFirst($r2);

// Move Root #2 with its child as first child of Root #2 - above "C".
$root2->moveAsFirst($root1);

```

As result, the tree looks like:

```
ROOT A
    ROOT 2
        2.a
    C
    A
    B
```

**Comment by sergeyklay** [comment]
Fixed in `2.0.x`
